test/action_controller/namespace_lookup_test.rb

```diff
@@ -125,7 +125,7 @@ def namespace_set_by_request_headers
       tests Api::V3::LookupTestController
 
       setup do
-        @test_namespace = self.class.parent
+        @test_namespace = self.class.module_parent
```

Comment:
It could call `Api::V3::LookupTestController.new.namespace_for_serializer` I guess

Reply:
`Api::V3::LookupTestController.new.namespace_for_serializer # => ActionController::Serialization::NamespaceLookupTest::Api::V3` 

But

`self.class.parent # => ActionController::Serialization`. 

@bf4 Will it make any issue if we hardcode the namespace for this case only?

or we can define `namespace_for_class` in a global namespace so that we can call it from here?